[docs] Add docs page for unstyled popper (#31813)
mui/material-ui: docs/data/base/components/popper/popper.md

---
product: base
title: React Popper component
components: PopperUnstyled
githubLabel: 'component: Popper'
---

> # Popper unstyled

Reviewer: ```suggestion
# Unstyled popper
```

> <p class="description">A Popper can be used to display some content on top of another.</p>

Reviewer 2: ```suggestion
<p class="description">The Popper component lets you create components that are displayed on top of other elements.</p>
```

To follow the pattern we're establishing with Modal and input.

> Some important acknowledgements of the Popper component:

Author: Here is the updated version, what do you think?

<img width="800" alt="Screen Shot 2565-03-29 at 11 31 40" src="https://user-images.githubusercontent.com/18292247/160533653-b72ffcc9-fb8d-407a-832c-89be82413e0d.png">

> Some important acknowledgements of the Popper component:

Reviewer 2: I like it. But still a bit unsure about the bundle size emphasis... why are we highlighting this on the Popper component and not the other ones? 🤔

> Some important acknowledgements of the Popper component:

Author: @michaldudak should I remove it for now to be consistent?

> Some important acknowledgements of the Popper component:

Reviewer: I'm more inclined to add the bundle size to other components as well. But perhaps using a nicer widget, not just plain text. For now, let's leave it and we can update all components' docs in a separate PR.

> Some important acknowledgements of the Popper component:

Reviewer 2: Ok!

- 🕷 Popper relies on the 3rd party library ([Popper.js v2](https://popper.js.org/docs/v2/)) for perfect positioning.
- 💄 It's an alternative API to [react-popper](https://popper.js.org/react-popper/v2/). It aims for simplicity.
- 📦 [8 kB gzipped](/size-snapshot).

> - The children is the [`Portal`](/base/react-portal/) prepended to the body of the document to avoid rendering problems.

Reviewer: ```suggestion
- The children are placed in a [`Portal`](/base/react-portal/) prepended to the body of the document to avoid rendering problems.
```

  You can disable this behavior with `disablePortal` prop.

> - The scroll isn't blocked when the popper open.

Reviewer: ```suggestion
- The page scroll isn't blocked when the popper opens.
```

- The placement of the popper updates with the available area in the viewport.
- Clicking away does not hide the `Popper` component. If you need this behavior, you can use [`ClickAwayListener`](/base/react-click-away-listener/)
- The `anchorEl` is passed as the reference object to create a new `Popper.js` instance.

{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}

## Basic Popper

> By default, the popper is mounted to the DOM when it receives `open` prop as `true` and is removed from the DOM when `open` is `false`. If you want to implement a transition, use `keepMounted` prop (see [Transition](#transition) example below).

Reviewer 2: ```suggestion
By default, the popper is mounted to the DOM when it receives `open` prop as `true` and then is removed from it when `open` turns `false`. If you want that to happen with a transition, use the `keepMounted` prop (see the [Transition](#transition) example below).
```

---
product: base
title: React Popper component
components: PopperUnstyled
githubLabel: 'component: Popper'
---

# Unstyled popper

<p class="description">The Popper component lets you create components that are displayed on top of other elements.</p>

**Caveats**:

- 🕷 Popper relies on the 3rd party library ([Popper.js v2](https://popper.js.org/docs/v2/)) for perfect positioning.
- 💄 It's an alternative API to [react-popper](https://popper.js.org/react-popper/v2/). It aims for simplicity.

> - Clicking away does not hide the `Popper` component. If you need this behavior, you can use [`ClickAwayListener`](/base/react-click-away-listener/)

Reviewer: I'd say this is the only "caveat". The other bullet points should be under "features"

Author: Done!

- The `anchorEl` is passed as the reference object to create a new `Popper.js` instance.

**Features**:

- The children are placed in a [`Portal`](/base/react-portal/) prepended to the body of the document to avoid rendering problems.
  You can disable this behavior with `disablePortal` prop.
- The page scroll isn't blocked when the popper opens.
- The placement of the popper updates with the available area in the viewport.

**Bundle size**: 📦 [8 kB gzipped](/size-snapshot).

{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}

## Basic Popper

> By default, the popper is mounted to the DOM when it receives `open` prop as `true` and then is removed from it when `open` turns `false`. If you want that to happen with a transition, use the `keepMounted` prop (see the [Transition](#transition) example below).

Reviewer: ```suggestion
By default, the popper is mounted to the DOM when its `open` prop is set to `true` and removed from it when `open` turns `false`. If you want that to happen with a transition, use the `keepMounted` prop (see the [Transition](#transition) example below).
```

Author: updated!
